compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/environment/EnvironmentPackageCache.java

```diff
@@ -20,65 +20,75 @@
  *
  * @since 2.0.0
  */
-public class EnvironmentPackageCache implements WritablePackageCache {
+public class EnvironmentPackageCache {
 
-    private final Map<PackageId, Project> projects = new HashMap<>();
+    private final Map<PackageOrg, Map<PackageName, Map<PackageVersion, Package>>> projects;
 
-    public void cache(Package pkg) {
-        projects.put(pkg.packageId(), pkg.project());
+    public EnvironmentPackageCache() {
+        this.projects = new HashMap<>();
     }
 
-    @Override
-    public Optional<Package> getPackage(PackageId packageId) {
```

Comment:
Since this is a public API, there can be users who are already using this code. Removing the method can break existing code.

Reply:
You want me to overload the function instead?

Reply:
I'm assuming it'll be the same for all the other functions Ive replaced too right?

Reply:
or do you want me to replace the implementations of `getPackage()` throughout the code?
Else I would have to define the variable `projects` in `EnvironmentalPackageCache` twice once with a hashmap with PackageId and one without

Reply:
Below is the parent interface from which all caches are implemented on:
https://github.com/ballerina-platform/ballerina-lang/blob/b23b120418343cfe812b457749e2126143a1d5dc/compiler/ballerina-lang/src/main/java/io/ballerina/projects/environment/PackageCache.java#L34

Here is what you can do to this interface:
1. Deprecate the APIs that accept the PackageId
2. Introduce alternative APIs to accept the `PackageOrg`, `PackageName`, and `PackageVersion`

Replacing the usage of the deprecated APIs would be the next step which will be cleaner to address in another PR.